### PR TITLE
mpd: deleted hardcoded config file path from plist and added a caveat

### PR DESCRIPTION
**Pull request by jerryskye**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Hey guys [#30683](https://github.com/Homebrew/homebrew-core/issues/30683) sorry about that last [PR](https://github.com/Homebrew/homebrew-core/pull/29344), now when I look at it it's obvious I wasn't thinking then. Apologize for all the mess. Have a nice day :wink:

### CODE REVIEW

**Comment by Hamuko** [comment]
`~/.mpd/mpd.conf` is also one possible location for the default configuration file. I personally use that one since I can just cram all of MPD's files inside the same `~/.mpd/` path.

File: Formula/mpd.rb [hunk 1]
```diff
@@ -112,7 +112,17 @@ def install
     (etc/"mpd").install "doc/mpdconf.example" => "mpd.conf"
   end
 
-  plist_options :manual => "mpd #{HOMEBREW_PREFIX}/etc/mpd/mpd.conf"
+  def caveats; <<~EOS
+    MPD requires a config file to start. Please copy the default config from #{etc}/mpd/mpd.conf into one of these paths:
+      - $XDG_CONFIG_HOME/mpd/mpd.conf
+      - ~/.mpd/mpd.conf
+      - ~/.mpdconf
+      - /etc/mpd.conf
```
Comment (DomT4) [inline Formula/mpd.rb]: I'd rather we not encourage anyone to touch `/etc`.

File: Formula/mpd.rb [hunk 2]
```diff
@@ -112,7 +112,16 @@ def install
     (etc/"mpd").install "doc/mpdconf.example" => "mpd.conf"
   end
 
-  plist_options :manual => "mpd #{HOMEBREW_PREFIX}/etc/mpd/mpd.conf"
+  def caveats; <<~EOS
+    MPD requires a config file to start. Please copy the default config from #{etc}/mpd/mpd.conf into one of these paths:
```
Comment (DomT4) [inline Formula/mpd.rb]: Can we wrap this to 80 characters please.

**Comment by DomT4** [comment]
Is this likely to have caused enough breakage/confusion in the time it has existed to merit having everyone reinstall `mpv`?

**Comment by jgillman** [comment]
@DomT4 Possibly? MPD itself is totally fine, it's just the launchd service file that would need to be updated... If that requires a reinstall then yeah, anyone who installed/updated since Jun 24 would have to reinstall.

**Comment by DomT4** [comment]
Alright, let's leave it be for now on the `revision` front but that option is open to us if necessary.

Merged in https://github.com/Homebrew/homebrew-core/commit/7e236f3a2a81fd7bc2d8a72d3074a0870d4ce696. Thank you for your contribution to Homebrew @jerryskye; we appreciate it! :smiley_cat:

**Pull request closed without merging**